fix: always return promise if async (#2728)
markedjs/marked: docs/USING_PRO.md

<h2 id="walk-tokens">Walk Tokens : <code>walkTokens</code></h2>

> The walkTokens function gets called with every token. Child tokens are called before moving on to sibling tokens. Each token is passed by reference so updates are persisted when passed to the parser. The return value of the function is ignored but awaited if running marked in [`async`](#async) mode.

Reviewer: ```suggestion
The walkTokens function gets called with every token. Child tokens are called before moving on to sibling tokens. Each token is passed by reference so updates are persisted when passed to the parser. When [`async`](#async) mode is enabled, the return value is awaited. Otherwise the return value is ignored. 
```
